packages/common-all/src/store/ConfigStore.ts

```diff
@@ -0,0 +1,336 @@
+import { errAsync, okAsync, ResultAsync } from "neverthrow";
+import { DendronError, IDendronError } from "../error";
+import { URI, Utils } from "vscode-uri";
+import { ConfigReadOpts, IConfigStore } from "./IConfigStore";
+import { IFileStore } from "./IFileStore";
+import { CONSTANTS, StatusCodes } from "../constants";
+import { ConfigUtils, DeepPartial } from "../utils";
+import { DendronConfig, DendronConfigValue, DVault } from "../types";
+import { fromStr, toStr } from "../yaml";
+import { DConfigLegacy } from "../oneoff/ConfigCompat";
+import _ from "lodash";
+
+export class ConfigStore implements IConfigStore {
+  private _fileStore: IFileStore;
+  private _wsRoot: URI;
+  private _cachedDendronConfig: DendronConfig | undefined;
+  private _homeDir: URI | undefined;
+
+  get path(): URI {
+    return Utils.joinPath(this._wsRoot, CONSTANTS.DENDRON_CONFIG_FILE);
+  }
+
+  constructor(fileStore: IFileStore, wsRoot: URI, homeDir: URI | undefined) {
+    this._fileStore = fileStore;
+    this._wsRoot = wsRoot;
+    this._homeDir = homeDir;
+  }
+
+  create(
```

Comment:
should we maybe name this `createConfig` instead? 

Maybe just a cause of our code base conventions, but reading this makes me think that it's some factory method of `ConfigStore`.

Reply:
good point. maybe all methods should have that appended.